core/services/ocrcommon/peer_wrapper.go

```diff
@@ -193,21 +197,25 @@ func (p *SingletonPeerWrapper) Start(context.Context) error {
 		if err != nil {
 			return errors.Wrap(err, "error calling NewPeer")
 		}
-		p.Peer = &peerAdapter{
+		p.Peer1 = &peerAdapterOCR1{
 			peer.OCR1BinaryNetworkEndpointFactory(),
 			peer.OCR1BootstrapperFactory(),
 		}
-		p.Peer2 = &peerAdapter2{
+		p.Peer2 = &peerAdapterOCR2{
 			peer.OCR2BinaryNetworkEndpointFactory(),
 			peer.OCR2BootstrapperFactory(),
 		}
+		p.peerCloser = peer
 		return nil
 	})
 }
 
 // Close closes the peer and peerstore
 func (p *SingletonPeerWrapper) Close() error {
 	return p.StopOnce("SingletonPeerWrapper", func() (err error) {
+		if !reflect.ValueOf(p.peerCloser).IsNil() {
+			err = p.peerCloser.Close()
+		}
```

Comment:
Was this still happening? I would guess that it could be left off with the way this ended up :shrug:

Reply:
Nothing is different about it aside from the interface implementing fewer functions.  Why would that make any difference?

Reply:
I mean that if this is happening then I would be very surprised, so I suspect it could be excluded. If not, then we should fix the root cause that creates this situation instead of trying to recognize if afterwards.